### ISSUE TITLE
session cache files are not deleted from server

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1118]
_Imported from trac ticket 1118.
Created by **kbower**
Opened in trac: Oct 7, 2011, 8:03:56 AM
Last modified in trac: Apr 17, 2012, 4:33:36 PM_
***
from https://beaker.groovie.org/sessions.html#deleting :
## Removing Expired/Old Sessions
Beaker does not automatically delete expired or old cookies on any of its back-ends. This task is left up to the developer based on how sessions are being used, and on what back-end.

The database backend records the last accessed time as a column in the database so a script could be run to delete session rows in the database that haven’t been used in a long time.

When using the file-based sessions, a script could run to remove files that haven’t been touched in a long time, for example (in the session’s data dir):

```
find . -mtime +3 -exec rm {} \;
```----
Also, find a way to move the cache files out of /trunk.  Probably need a symbolic link to wsgi-deployment.py from home directory or else find place to specify where /data directory should be written.

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 7, 2011, 3:10:27 PM:_

(Changeset [[1790]](../commit/31678c86b5603f48ae6316956fca60d7d8fafc3c))
 * move beaker session data from out of trunk to the rarch home directory (see #1118)

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 10, 2011, 8:19:39 AM:_

# Implementation direction
See http://groups.google.com/group/modwsgi/browse_thread/thread/c6a8d5eb8f821590?hl=en

Suppose I am running in a multiprocess environment but would like to
run a "scheduler".  Of course, each process starts its own scheduler,
so I end up with this running once for each process.

Can you think of a way around this?  I only want one process to start
the scheduler....

```python
import tgscheduler
import logging
log = logging.getLogger(__name__)

def kent():
   log.info("run")

tgscheduler.start_scheduler()
tgscheduler.add_interval_task(kent,15, initialdelay=30)
```
Any ideas?

Thanks,
Kent
----
Create a dummy daemon process group and preload a Python script file
into it to do the background thread. No URLs would ever be mapped to
the daemon process.
```
  WSGIDaemonProcess mysite processes=3 threads=3
  WSGIScriptAlias / /some/path/mysite.wsgi process-group=mysite

  WSGIDaemonProcess mytasks processes=1 threads=1
  WSGIImportScript /some/path/mytasks.py process-group=mytasks
application-group=%{GLOBAL}
```The threads=1 to mytasks is just to cut down on request threads that
aren't going to be used anyway.

Graham

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 10, 2011, 8:38:15 AM:_

See also [WSGICallableObject directive](http://code.google.com/p/modwsgi/wiki/ConfigurationDirectives#WSGICallableObject)

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 11, 2011, 7:59:37 AM:_

One more question: I would want this standalone process to load the same
application so that it has the same setup/model, etc.

Is there a Directive I can use to pass a parameter to the python script so I
can tell the "special" process it is allowed to load the scheduler?  Or
would I use WSGICallableObject to accomplish this?
----
Use:
```python
  import mod_wsgi

  if mod_wsgi.process_group == 'mytasks':
    ...
```IOW, conditional on name of process group.

Graham

**Comment by kbower** [comment]
_**kbower** trac comment on Apr 17, 2012, 4:33:36 PM:_

(Changeset [[2549]](../commit/6aa8c0b80938ca6e965e9d910d95706dee53486d))
 * database managed application locks
 * processes framework
   * restructure pylotengine/__init__.py so can import from pylotengine.lib (or other modules) without dependency problems caused by pylotengine/__init__.py
 * application views
   * heldlocks
   * runningprocesses
   * currentprocesses
 * sessionfilepurge.py process to purge /sessiondata files (closes #1118)
 * systemeventprocessor.py copies system events to subscriptionevents (also see [#1537](1537))

**Comment by kbower** [comment]
_**kbower** trac status change on Apr 17, 2012, 4:33:36 PM:_ **closed**
